scripts/analytics/addon_snapshot.py

```diff
@@ -82,9 +82,9 @@ def get_events(self, date=None):
 
             connected_count = 0
             for node_settings in node_settings_list:
-                if not node_settings.owner.is_bookmark_collection:
-                    connected_count += 1
-
+                connected_count += 1
+                if not node_settings.owner or node_settings.owner.is_bookmark_collection:
```

Comment:
What about instead:

```py
if node_settings.owner and not node_settings.owner.is_bookmark_collection:
    connected_count += 1
```

Or possibly getattr() might do what you need as well. Mostly the adding then subtracting thing bothered me, but then I started being annoyed with the `if object.exists and object.exists.subproperty` pattern and wanted to see the better way of doing that because there should be and there kind of is but generalizable blah rabbit hole blah rabbit hole.